### ISSUE TITLE
Keyset pagination - aggregate field must be included in query when sorting on that field

### ISSUE THREAD

**Issue by ahey** [ash-project/ash_graphql#74]
When performing a keyset paginated read that sorts on an aggregate field, an error occurs if that same aggregate field is not also included in the GQL query. The error **only occurs when loading the second page specifying an `after` value.**

Assuming that the `Post` resource has an aggregate field called `authorName`, this query to get the first page works as expected:

```graphql
query KeysetPaginatedPosts {
  keysetPaginatedPosts(
  sort: [{field: AUTHOR_NAME}]
  first: 1
) {
    count
    startKeyset
    endKeyset
    results{
      id
    }
  }
}
```

Taking the `endKeyset` value from the first query, and using it to request the second page causes the error to occur:

```graphql
query KeysetPaginatedPosts {
  keysetPaginatedPosts(
  sort: [{field: AUTHOR_NAME}]
  first: 1
  after: "keyset value from first query here"
) {
    count
    startKeyset
    endKeyset
    results{
      id
    }
  }
}
```

The error is:

```elixir
** (Ash.Error.Unknown) Unknown Error

* ** (Ecto.Query.CastError) deps/ash_postgres/lib/expr.ex:485: value `#Ash.NotLoaded<:aggregate>` in `where` cannot be cast to type #Ash.Type.StringWrapper.EctoType<[]> in query
```

The issue can be worked around by including the aggregate field name in the query, and then fetching the second page works as expected.

```graphql
query KeysetPaginatedPosts {
  keysetPaginatedPosts(
  sort: [{field: AUTHOR_NAME}]
  first: 1
) {
    count
    startKeyset
    endKeyset
    results{
      id
      authorName
    }
  }
}
```

**Comment by zachdaniel** [comment]
Thanks for the report, I'll take a look tomorrow!

**Comment by zachdaniel** [comment]
This should be solved in `ash` version `2.9.18` :) Its building from CI currently, and will be released shortly.

**Comment by ahey** [comment]
This is confirmed fixed in my codebase now. Thank you!
